slack-api-model/src/main/java/com/slack/api/util/thread/ExecutorServiceFactory.java

```diff
@@ -4,16 +4,10 @@
 import java.util.concurrent.Executors;
 import java.util.concurrent.ScheduledExecutorService;
 
-public class ExecutorServiceFactory {
+// Use ExecutorServiceProvider interface instead
+@Deprecated
```

Comment:
Does this class get exposed to developers at all?

Reply:
@filmaj We don't encourage developers to use this class at all but technically it's possible to utilize this class if you want to do so because it's a public class. That's why I didn't immediately delete this class (for backward compatibility). Also, `DaemonThreadExecutorServiceFactory` is a more accurate name for it. Thus, I created a super class with the same set of functionalities and new code inside the SDK relies on only super class.